elements/dash-tooltip/dash-tooltip.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * A tooltip is a popup that displays information related to an element
+ * when the element receives keyboard focus or the mouse hovers over it.
+ * It typically appears after a small delay and disappears when Escape is
+ * pressed or on mouse out. The element that triggers the tooltip references
+ * the tooltip element with aria-describedby.
+ * See: https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip
+ */
+
+class DashTooltip extends HTMLElement {
+  /**
+  * The constructor does work that needs to be executed _exactly_ once.
+  */
+  constructor() {
+    super();
+
+    this._show = this._show.bind(this);
+    this._hide = this._hide.bind(this);
+  }
+
+  /**
+   * `connectedCallback` is a life cycle callback defined by CustomElements v1
+   * that gets called when an instance of the element gets inserted into the
+   * DOM. Theoretically, `connectedCallback` could be called multiple
+   * times when an element gets moved around.
+   */
+  connectedCallback() {
+    // A tooltip should always set its `role` to `tooltip`
+    this.setAttribute('role', 'tooltip');
+
+    // Tooltips cannot be focused themselves.
+    this.setAttribute('tabindex', -1);
```

Comment:
@surma @notwaldorf just curious, since `tabIndex` can also be set as a property, do we want to standardize on one approach?